torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -295,12 +297,21 @@ class StateDictType(Enum):
                these APIs are meant for use via the :func:`state_dict_type`
                context manager as follows:
                    >>> # xdoctest: +SKIP("undefined variables")
-                   >>> with fsdp.state_dict_type(StateDictType.LOCAL_STATE_DICT):
-                   ...     state = fsdp.state_dict()  # loads local state dict
+                   >>> with FSDP.state_dict_type(StateDictType.LOCAL_STATE_DICT):
+                   >>>     state = model.state_dict()
             3. ``_sharded_state_dict/_load_sharded_state_dict``: this pair of APIs
                return and load sharded, unflattened parameters. The ``state_dict``
                return by ``sharded_state_dict`` can be used by all other parallel
                schemes (resharding may be required).
+    ..note::
+        The StateDictType also controls how FSDP.optim_state_dict() return the
+        states of the optimizer. The sharding strategy will be consistent for
+        ``model.state_dict()` and ``FSDP.optim_state_dict()``:
+            >>> with FSDP.state_dict_type(StateDictType.SHARDED_STATE_DICT):
```

Comment:
do we anticipate a use case where one state_dict is sharded and other is full, for some reason?